crates/rethnet_eth/src/signature.rs

```diff
@@ -93,6 +101,31 @@ impl fmt::Display for Signature {
 }
 
 impl Signature {
+    /// to obtain message_hash consider rethnet_eth::src::utils::hash_message().
+    pub fn new<M>(message: M, private_key: &SecretKey) -> Result<Self, secp256k1::Error>
```

Comment:
Because you removed the `FromStr` conversion, this is now infallible and doesn't require a `Result` type anymore.

```suggestion
    pub fn new<M>(message: M, private_key: &SecretKey) -> Self
```